### PR TITLE
conLev_3

### LINKED ISSUE #210: conLev_3
Description: Adding text alongside color bar to better explain the range of each contour level

NCL Script: https://www.ncl.ucar.edu/Applications/Scripts/conLev_3.ncl
NCL Plot: https://www.ncl.ucar.edu/Applications/Images/conLev_3_lg.png

Category: Contours

### PR DESCRIPTION
**Pull request by hCraker**
Closes #210 
image

### CODE REVIEW

File: Plots/Contours/NCL_conLev_3.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+# Use geocat.viz.util convenience function to add minor and major tick lines
+gvutil.add_major_minor_ticks(ax, x_minor_per_major=3, y_minor_per_major=5,
+                             labelsize=12)
+
+# Remove ticks on right side
+ax.tick_params(which='both', right=False)
+
+# Use geocat.viz.util convenience function to add title
+gvutil.set_titles_and_labels(ax, maintitle="Explaination of Python contour levels")
```
Comment (michaelavs) [inline Plots/Contours/NCL_conLev_3.py:83]: typo: "explanation"

File: Plots/Contours/NCL_conLev_3.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+"""
+
+###############################################################################
+# Import packages:
+import numpy as np
+import xarray as xr
+import matplotlib.pyplot as plt
+
+import geocat.datafiles as gdf
+from geocat.viz import cmaps as gvcmaps
```
Comment (michaelavs) [inline Plots/Contours/NCL_conLev_3.py:31]:
```suggestion

```
Unused line

File: Plots/Contours/NCL_conLev_3.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+cbar_ticks = np.arange(248, 308, 4)
+cbar = plt.colorbar(temp, orientation='vertical', pad=0.005)
+cbar.ax.tick_params(labelsize=11)
+cbar.set_ticks(cbar_ticks)
+
+# Use geocat.viz.util convenience function to set axes tick values
+gvutil.set_axes_limits_and_ticks(ax, xlim=(-140, -50), ylim=(20, 60),
+                                 xticks=[-135, -90],
+                                 yticks=np.arange(20, 70, 10))
+
```
Comment (michaelavs) [inline Plots/Contours/NCL_conLev_3.py:71]:
```suggestion
# Remove the degree symbol from tick labels
    ax.yaxis.set_major_formatter(LatitudeFormatter(degree_symbol=''))
    ax.xaxis.set_major_formatter(LongitudeFormatter(degree_symbol=''))
```

File: Plots/Contours/NCL_conLev_3.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+ax = plt.axes()
+
+# Import an NCL colormap
+newcmp = 'plasma'
+
+# Contourf-plot data (for filled contours)
+num_lev = 16  # Number of levels
+temp = T.plot.contourf(ax=ax, vmin=244, vmax=308, levels=np.linspace(244, 308, num_lev + 1),
+                       cmap=newcmp, add_colorbar=False, add_labels=False)
+# Contour-plot data (for line contours)
```
Comment (michaelavs) [inline Plots/Contours/NCL_conLev_3.py:57]:
```suggestion

# Contour-plot data (for line contours)
```
Added white space for clarity

File: Plots/Contours/NCL_conLev_3.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+text = '{} <= T < {}'
+for i in range(0, 14):
+    y = y + 1 / num_lev  # Vertical spacing between the labels
+    ax.text(0.904, y, text.format(cbar_ticks[i], cbar_ticks[i + 1]),
+            fontsize=size, horizontalalignment='center',
+            verticalalignment='center', transform=ax.transAxes,
+            bbox=dict(boxstyle='square, pad=0.25',
+                      facecolor='papayawhip',
+                      edgecolor='papayawhip'))
+y = y + 1 / num_lev  # Increment height once more for top label
```
Comment (michaelavs) [inline Plots/Contours/NCL_conLev_3.py:106]:
```suggestion

y = y + 1 / num_lev  # Increment height once more for top label
```
Added white space for clarity

File: Plots/Contours/NCL_conLev_3.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
+"""
+NCL_conLev_4.py
```
Comment (erogluorhan) [inline Plots/Contours/NCL_conLev_3.py:2]: typo, should be NCL_conLev_3, not 4.

**Review by erogluorhan — CHANGES_REQUESTED** [review]
Code and plot are great! Correct a little typo mentioned in line please

**Review by erogluorhan — APPROVED** [review]

**Pull request merged**